metis-remove-data/src/main/java/eu/europeana/metis/remove/discover/ExecutionPluginForest.java

```diff
@@ -57,39 +63,71 @@
 class ExecutionPluginForest {
 
   private final Map<String, ExecutionPluginNode> nodesById = new HashMap<>();
-  private final Map<PluginLinkByTypeAndTimestamp, ExecutionPluginNode> nodesByTypeAndTimestamp;
   private final List<ExecutionPluginNode> rootNodes;
   private final Set<String> leafNodeIds;
 
-  ExecutionPluginForest(List<WorkflowExecution> executions) {
-    this(executions, Collections.emptySet());
+  ExecutionPluginForest(List<WorkflowExecution> executions, boolean attemptToIgnoreDeletedPlugins) {
+    this(executions, attemptToIgnoreDeletedPlugins, Collections.emptySet());
   }
 
-  ExecutionPluginForest(List<WorkflowExecution> executions, Set<String> ignoredPluginIds) {
+  ExecutionPluginForest(List<WorkflowExecution> executions, boolean attemptToIgnoreDeletedPlugins,
+          Set<String> ignoredPluginIds) {
 
-    // Convert all executions and plugins to nodes and add them to the node map
+    // Convert all executions and plugins to nodes and add them to the node map, depending on
+    // whether they are to be ignored or not. No nodes are linked at this point.
+    final  Map<String, ExecutionPluginNode> ignoredNodesById = new HashMap<>();
     for (WorkflowExecution execution : executions) {
       for (AbstractMetisPlugin plugin : execution.getMetisPlugins()) {
-        if (!ignoredPluginIds.contains(plugin.getId())) {
-          final ExecutionPluginNode node = new ExecutionPluginNode(execution, plugin);
-          nodesById.put(node.getId(), node);
-        }
+        final boolean ignoreNode = ignoredPluginIds.contains(plugin.getId())
+                || (attemptToIgnoreDeletedPlugins && plugin.getDataStatus() == DataStatus.DELETED);
```

Comment:
Just for my understanding we are ignoring DELETED plugins because they are already deleted sometime before?
And does that give us much benefit over the case were we simply wouldn't ignore them and when we use them to delete data it simply won't do anything because there is no data in the db?

Reply:
Yes, it would. This is an iterative process. So, for instance, if this time we mark a media processing run as deleted (i.e. delete it from ecloud and give the plugin the status DELETED), that means that the next iteration perhaps we can delete the enrichment that preceded it (and now no longer has a non-deleted child). So it is quite important to ignore the deleted plugins here, otherwise we can never make progress. Does that make sense/answer your question?

Reply:
Yes I understand about the implementation and iteration in this method but I was referring to the general reason of checking the DELETED status, same like the next comment actually.
(After our call this is cleared out)